### PR TITLE
Code rearrangement

### PR DESCRIPTION
**Pull request by bell07**
This is a non-functional change. It is a rearrangement of code to match more the documented order in API-doc. I did it during the development because I had lost me in the code sometimes.
By the way I enhanced the code block header comments by short description from API.md. 
Pls. note, the callbacks minetest.register_on_player_receive_fields() and minetest.after() aren't in the doc

### CODE REVIEW

File: smartfs.lua [hunk 1]
```diff
@@ -41,14 +44,21 @@ function smartfs.create(name, onload)
 	return smartfs._fdef[name]
 end
 
-function smartfs.override_load_checks()
-	smartfs._loaded_override = true
-end
+------------------------------------------------------
+-- Smartfs Interface - Creates a new element type. (private)
```
Comment (rubenwardy) [inline smartfs.lua]: this is a public API as users should be able to define elements they want

File: smartfs.lua [hunk 2]
```diff
@@ -111,8 +105,176 @@ function smartfs.add_to_inventory(form, icon, title)
 	end
 end
 
+------------------------------------------------------
+-- Smartfs Interface - Returns the name of an installed and supported inventory mod that will be used above, or null.
```
Comment (rubenwardy) [inline smartfs.lua]: nil, not null

**Comment by bell07** [comment]
Sorry, I had trouble with git command line oO

**Comment by rubenwardy** [comment]
Looks good to me. Are you done with this?

**Comment by bell07** [comment]
Yes, I am done. The next PR's are waiting for this is merged ;-) But please take a small review for copy@paste things or something like

**Comment by bell07** [comment]
a kindly push. Do you need something more to get this PR into upstream?

**Pull request merged**